### PR TITLE
Added missing arduino definitions to nRF5340 Audio DK board

### PR DESCRIPTION
**Pull request by smaerup**

### CODE REVIEW

**Comment by nordicjm** [comment]
@koffes please review

**Review by koffes — APPROVED** [review]

**Review by larsgk — DISMISSED** [review]

**Comment by koffes** [comment]
Did anything change apart from a rebase and a rename in the last two force pushed @smaerup ?

**Comment by smaerup** [comment]
> Did anything change apart from a rebase and a rename in the last two force pushed @smaerup ?
No, that is the only changes, the rename was needed because of the HWM2 switch.

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi [hunk 1]
```diff
@@ -120,6 +131,23 @@ arduino_serial: &uart1 {
 	pinctrl-names = "default", "sleep";
 };
 
+arduino_i2c: &i2c1 {
+	compatible = "nordic,nrf-twim";
+	status = "okay";
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-1 = <&i2c1_sleep>;
+	pinctrl-names = "default", "sleep";
+};
```
Comment (anangl) [inline boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:140]: Instead of this just add `arduino_i2c:` in this line: https://github.com/zephyrproject-rtos/zephyr/blob/708b38d7f5d8dc81a73a90fc05e80a673beea51e/boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi#L151

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi [hunk 2]
```diff
@@ -120,6 +131,23 @@ arduino_serial: &uart1 {
...
+
+arduino_spi: &spi4 {
```
Comment (anangl) [inline boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:142]: https://github.com/zephyrproject-rtos/zephyr/blob/708b38d7f5d8dc81a73a90fc05e80a673beea51e/boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi#L203

**Review by anangl — CHANGES_REQUESTED** [review]

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi (hunk 1 above)
Comment (smaerup) [reply boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:140]: Good point, will fix.

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi (hunk 2 above)
Comment (smaerup) [reply boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:142]: Good point, will fix.

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi [hunk 3]
```diff
@@ -172,10 +183,11 @@ arduino_serial: &uart1 {
 	};
 };
 
-&spi4 {
+arduino_spi: &spi4 {
 	compatible = "nordic,nrf-spim";
 	status = "okay";
-	cs-gpios = <&gpio0 11 GPIO_ACTIVE_LOW>, <&gpio0 17 GPIO_ACTIVE_LOW>;
+	cs-gpios = <&gpio0 11 GPIO_ACTIVE_LOW>, <&gpio0 17 GPIO_ACTIVE_LOW>,
+	           <&arduino_header 16 GPIO_ACTIVE_LOW>;
```
Comment (anangl) [inline boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:190]:
```suggestion
	cs-gpios = <&arduino_header 16 GPIO_ACTIVE_LOW>,
	           <&gpio0 11 GPIO_ACTIVE_LOW>, <&gpio0 17 GPIO_ACTIVE_LOW>;
```
and adjust addresses in `sdhc@0` and `cs47l63@1` below. `arduino_spi` users (e.g. shields) will likely use CS #0, not #2.

Comment (larsgk) [reply boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:190]: Hi @anangl - could you please explain to a 'devicetree novice' like me? :)

**Review by anangl — DISMISSED** [review]
Please fix the compliance check failure (use tab instead of 8 spaces).

**Review by larsgk — DISMISSED** [review]

**Review by anangl — DISMISSED** [review]

**Review by koffes — APPROVED** [review]

**Review by larsgk — DISMISSED** [review]
The previous merge issue seems fixed now

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi [hunk 4]
```diff
@@ -137,23 +137,6 @@ arduino_i2c: &i2c1 {
 	pinctrl-0 = <&i2c1_default>;
 	pinctrl-1 = <&i2c1_sleep>;
 	pinctrl-names = "default", "sleep";
-};
-
-arduino_spi: &spi4 {
```
Comment (anangl) [inline boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi]: You should correct the commits. Currently, the second commit removes changes in `boards/` that are just unnecessarily introduced by the first one.

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi [hunk 5]
```diff
@@ -172,14 +183,15 @@ arduino_serial: &uart1 {
...
 	compatible = "nordic,nrf-spim";
 	status = "okay";
-	cs-gpios = <&gpio0 11 GPIO_ACTIVE_LOW>, <&gpio0 17 GPIO_ACTIVE_LOW>;
+	cs-gpios = <&arduino_header 16 GPIO_ACTIVE_LOW>,
+		<&gpio0 11 GPIO_ACTIVE_LOW>, <&gpio0 17 GPIO_ACTIVE_LOW>;
 	pinctrl-0 = <&spi4_default>;
 	pinctrl-1 = <&spi4_sleep>;
 	pinctrl-names = "default", "sleep";
-	sdhc0: sdhc@0 {
+	sdhc0: sdhc@1 {
```
Comment (koffes) [inline boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:194]: Why was this changed to 1?

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi [hunk 6]
```diff
@@ -191,7 +203,7 @@ arduino_serial: &uart1 {
 		spi-max-frequency = <8000000>;
 	};
 
-	cs47l63: cs47l63@1 {
+	cs47l63: cs47l63@2 {
```
Comment (koffes) [inline boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:206]: Why was this changed to 2?

**Review by anangl — APPROVED** [review]

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi (hunk 5 above)
Comment (smaerup) [reply boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:194]: It was suggested by anangl above that shield users will probably use CS0, hence sdhc and cs47l63 should be incremented by 1.

File: boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi (hunk 6 above)
Comment (smaerup) [reply boards/nordic/nrf5340_audio_dk/nrf5340_audio_dk_nrf5340_cpuapp_common.dtsi:206]: It was suggested by anangl above that shield users will probably use CS0, hence sdhc and cs47l63 should be incremented by 1.

**Review by larsgk — APPROVED** [review]
